pkg/plugins/policies/core/matchers/validators/validator.go

```diff
@@ -34,18 +41,62 @@ func ValidateTargetRef(
 		case common_api.MeshSubset:
 			verr.Add(disallowedField("name", ref.Name, refKind))
 			verr.Add(disallowedField("mesh", ref.Mesh, refKind))
-		case common_api.MeshService:
+			verr.Add(validTags(ref.Tags))
+		case common_api.MeshService, common_api.MeshHTTPRoute:
 			verr.Add(requiredField("name", ref.Name, refKind))
+			verr.Add(validName(ref.Name))
 			verr.Add(disallowedField("mesh", ref.Mesh, refKind))
 			verr.Add(disallowedField("tags", ref.Tags, refKind))
-		case common_api.MeshServiceSubset:
+		case common_api.MeshServiceSubset, common_api.MeshGateway:
 			verr.Add(requiredField("name", ref.Name, refKind))
+			verr.Add(validName(ref.Name))
 			verr.Add(disallowedField("mesh", ref.Mesh, refKind))
+			verr.Add(validTags(ref.Tags))
 		}
 	}
 	return verr
 }
 
+func validTags(tags map[string]string) validators.ValidationError {
```

Comment:
we have a convention to name these kind of functions with `validate` prefix
```suggestion
func validateTags(tags map[string]string) validators.ValidationError {
```

Reply:
Done 👍